crates/handlers/src/oauth2/token.rs

```diff
@@ -343,16 +331,20 @@ async fn authorization_code_grant(
             claims::AUTH_TIME.insert(&mut claims, last_authentication.created_at)?;
         }
 
-        claims::AT_HASH.insert(&mut claims, hash(Sha256::new(), &access_token_str)?)?;
-        claims::C_HASH.insert(&mut claims, hash(Sha256::new(), &grant.code)?)?;
-
         let alg = client
             .id_token_signed_response_alg
             .unwrap_or(JsonWebSignatureAlg::Rs256);
         let key = key_store
             .signing_key_for_algorithm(alg)
             .context("no suitable key found")?;
 
+        if let Some(hash) = hash_token(alg, &access_token_str)? {
+            claims::AT_HASH.insert(&mut claims, hash)?;
+        }
+        if let Some(hash) = hash_token(alg, &grant.code)? {
+            claims::C_HASH.insert(&mut claims, hash)?;
+        }
```

Comment:
Here it will never try to sign with the `one` alg: the key store will throw an error because no key will match the `none` alg

Reply:
Right, because the `none` alg is not supported by the server.